### PR TITLE
People 125 user skills page  rake task

### PR DESCRIPTION
**Pull request by matDobek**
# Ticket
https://netguru.atlassian.net/browse/PEOPLE-125

# Description
- Adding rake task to assign users to default skills
- Add validating uniqueness of skill in user scope (given user, can only have 1  given skill)
- changed `app/services/skills/user_skill_rates_generator.rb` to `find_or_create`, instead of `create` (to handle case, when user got already assigned skills)

### Before merge checklist
- [x] CircleCI is passing
- [ ] Proper labels are set

### CODE REVIEW

File: spec/models/user_skill_rate_spec.rb [hunk 1]
```diff
@@ -4,9 +4,53 @@
...
+
+      context 'when skill is already assigned to user' do
+        it 'raises validation error' do
+          create(:user_skill_rate, user: user, skill: skill)
+          user_skill_rate = build(:user_skill_rate, user: user, skill: skill)
+
+          user_skill_rate.valid?
+
+          result = user_skill_rate.errors.full_messages
+          expect(result).to include("Skill has already been taken")
```
Comment (netguru-hound) [inline spec/models/user_skill_rate_spec.rb]: Prefer single-quoted strings when you don't need string interpolation or special symbols.

File: spec/lib/tasks/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  end
+
+  describe 'users:generate_default_user_skills' do
+    let(:run_rake_task) do
+      Rake::Task['users:generate_default_user_skills'].reenable
+      Rake.application.invoke_task 'users:generate_default_user_skills'
+    end
+
+    it 'generates skills for users' do
+      user = create(:user)
```
Comment (DariuszMusielak) [inline spec/lib/tasks/users_spec.rb]: Please use `let`

File: spec/lib/tasks/users_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      rated_skill = create(:skill)
+      non_assigned_skill = create(:skill)
+
+      create(:user_skill_rate, skill: rated_skill, user: user, rate: 1)
+
+      run_rake_task
```
Comment (DariuszMusielak) [inline spec/lib/tasks/users_spec.rb]: `subject`?

File: spec/models/user_skill_rate_spec.rb [hunk 4]
```diff
@@ -4,9 +4,53 @@
...
+    it { is_expected.to  validate_presence_of :skill }
+    it { is_expected.to  validate_presence_of :user }
+
+    describe 'uniqueness of :skill in given :user scope' do
+      let(:user) { create(:user) }
+      let(:another_user) { create(:user) }
+      let(:skill) { create(:skill) }
+      let(:another_skill) { create(:skill) }
+
+      context 'when skill is already assigned to user' do
```
Comment (DariuszMusielak) [inline spec/models/user_skill_rate_spec.rb]: Please use `let` and `subject`

File: spec/lib/tasks/users_spec.rb (hunk 2 above)
Comment (matDobek) [inline spec/lib/tasks/users_spec.rb]: 👍

File: spec/lib/tasks/users_spec.rb (hunk 3 above)
Comment (matDobek) [inline spec/lib/tasks/users_spec.rb]: actually I think that `run_rake_task` is better suited than subject
-> we want to signal: `hey, now we're running rake task, beware!`
-> it's common template for majority of rake tasks to be written this way ( https://blog.pivotal.io/labs/labs/how-i-test-rake-tasks )

File: spec/models/user_skill_rate_spec.rb (hunk 4 above)
Comment (matDobek) [inline spec/models/user_skill_rate_spec.rb]: 👍

**Pull request merged**